### PR TITLE
tests: integration tests for the Gitea migration form

### PR DESCRIPTION
**Pull request by realaravinth**
This PR originates from [@dachary](https://lab.forgefriends.org/dachary) of the [ForgeFriends Project](https://forgefriends.org). Please see [here](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/52) for origin.

### CODE REVIEW

File: integrations/migrate_test.go [hunk 1]
```diff
@@ -40,3 +45,54 @@ func TestMigrateLocalPath(t *testing.T) {
...
+
+		// Step 0: verify the repo is available
+		req := NewRequestf(t, "GET", fmt.Sprintf("/%s/%s", ownerName, repoName))
+		_ = session.MakeRequest(t, req, http.StatusOK)
+		// Step 1: get the Gitea migration form
+		req = NewRequestf(t, "GET", "/repo/migrate/?service_type=%d", structs.GiteaService)
+		resp := session.MakeRequest(t, req, http.StatusOK)
+		// Step 2: load the form
+		htmlDoc := NewHTMLParser(t, resp.Body)
+		link, exists := htmlDoc.doc.Find("form.ui.form[action^=\"/repo/migrate\"]").Attr("action")
```
Comment (Gusted) [inline integrations/migrate_test.go]:
```suggestion
		link, exists := htmlDoc.doc.Find(`form.ui.form[action^="/repo/migrate"]`).Attr("action")
```

**Review by Gusted — COMMENTED** [review]
So why are we specifically testing the HTML way of migrating and not via the API?

**Comment by realaravinth** [comment]
@gusted thanks for your review. This test is for the HTML form because it is a different codepath than the API, even though they both eventually lead to the same migration code.

---
[src](https://lab.forgefriends.org/forgefriends/forgefriends/-/merge_requests/52#note_1438)

**Review by Gusted — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
.-

**Pull request merged**